### PR TITLE
New azure storage integration

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/27981

## Description
New integration for azure storage

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    name: azure-storage-auth-reset
+  - arguments:
+    - default: false
+      isArray: false
+      name: account_name
+      required: false
+      secret: false
+    deprecated: false
+    execution: false
+    name: azure-storage-account-list
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:63]: I think you are missing command, argument and context descriptions - please add them in all the yml (can be done through the demisto-settings section in the pycharm plugin.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+      isArray: false
+      name: account_name
+      required: true
+      secret: false
+    deprecated: false
+    execution: false
+    name: azure-storage-blob-service-properties-get
+  - deprecated: false
+    execution: false
+    name: azure_storage_blob_service_properties_set
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:209]:
```suggestion
    name: azure-storage-blob-service-properties-set
```

I know you are yet to implement this but just note this.

**Review by roysagi — CHANGES_REQUESTED** [review]
Great first draft - Please see my notes and we can discuss them later.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+def storage_account_list(client: AKSClient, args: Dict) -> CommandResults:
+    account_name = args.get('account_name', '')
+    response = client.storage_account_list_request(account_name)
+    accounts = response.get('value', [response])
+    readable_output = [{
+        'Account Name': account.get('name'),
+        'Subscription ID': re.search('subscriptions/(.+?)/resourceGroups', account.get('id')).group(1),
+        'Resource Group': re.search('resourceGroups/(.+?)/providers', account.get('id')).group(1),
+        'Kind': account.get('kind'),
+        'Sku': account.get('sku', {}),
```
Comment (Itay4) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py:138]:
```suggestion
        'Sku': account.get('sku'),
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+    }
+    if args.get('tags'):
+        account_args['tags'] = args.get('tags')
+    if args.get('custom_domain_name'):
+        account_args['custom_domain_name'] = args.get('custom_domain_name')
+        if args.get('use_sub_domain_name'):
+            account_args['use_sub_domain_name'] = argToBoolean(args.get('use_sub_domain_name'))
+        else:
+            account_args['use_sub_domain_name'] = False
+    if args.get('enc_key_source'):
```
Comment (Itay4) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py:172]: use the walrus operator in all these conditions

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+def azure_storage_blob_service_properties_set(client: AKSClient, args: Dict):
+    return
+
+
+def start_auth(client: AKSClient) -> CommandResults:
+    user_code = client.ms_client.device_auth_request()
+    return CommandResults(readable_output=f"""### Authorization instructions
+1. To sign in, use a web browser to open the page [https://microsoft.com/devicelogin](https://microsoft.com/devicelogin)
+ and enter the code **{user_code}** to authenticate.
+2. Run the **!azure-ks-auth-complete** command in the War Room.""")
```
Comment (Itay4) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py:233]:
```suggestion
2. Run the **!azure-storage-auth-complete** command in the War Room.""")
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+
+
+def test_connection(client: AKSClient) -> str:
+    client.ms_client.get_access_token()
+    return '✅ Success!'
+
+
+def reset_auth() -> str:
+    set_integration_context({})
+    return 'Authorization was reset successfully. Run **!azure-ks-auth-start** to start the authentication process.'
```
Comment (Itay4) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py:248]:
```suggestion
    return 'Authorization was reset successfully. Run **!azure-storage-auth-start** to start the authentication process.'
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
+description: Deploy and manage containerized applications with a fully managed Kubernetes
+  service.
+display: Azure Storage (Beta)
```
[the remark this answers is not in the document]
Comment (Itay4) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:32]: we're using stable API here, i don't think this should be beta

[the remark this answers is not in the document]
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:32]: this is what the design says, do remove anyway?

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+            'sku': {
+                'name': sku
+            },
+            'kind': kind,
+            'location': location,
+            'properties': {}
+        }
+        if tags is not None:
+            args_tags_list = tags.split(',')
+            tags_obj = {f'tag{str(i+1)}': args_tags_list[i] for i in range(len(args_tags_list))}
```
[the remark this answers is not in the document]
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py:85]: yes, it's display in the request body example.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py [hunk 9]
```diff
@@ -0,0 +1,291 @@
+import demistomock as demisto
```
[the remark this answers is not in the document]
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py:1]: done

[the remark this answers is not in the document]
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py:1]: done

**Review by roysagi — CHANGES_REQUESTED** [review]
Great job @yaakovpraisler! - 
Please fix my few comments and then add a test playbook and go ahead and output a readme for this - you can read more about this here: https://xsoar.pan.dev/docs/documentation/readme_file

File: Packs/AzureStorage/Integrations/AzureStorage/README.md [hunk 10]
```diff
@@ -0,0 +1,5 @@
+This README contains the full documentation for your integration.
```
[the remark this answers is not in the document]
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/README.md:1]: After the demos? or before

[the remark this answers is not in the document]
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/README.md:1]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py (hunk 3 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py (hunk 4 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py (hunk 5 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py (hunk 6 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/README.md [hunk 11]
```diff
@@ -0,0 +1,637 @@
+Deploy and manage storage accounts and blob services.
+This integration was integrated and tested with version 2019-06-01 of Azure Storage
+## Permissions
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/README.md:3]: Please fill this section with the required permissions if necessary.

File: Packs/AzureStorage/Integrations/AzureStorage/README.md [hunk 12]
```diff
@@ -0,0 +1,637 @@
...
+
+    | **Parameter** | **Required** |
+    | --- | --- |
+    | Application ID | True |
+    | Subscription ID | True |
+    | Resource Group Name | True |
+    | Trust any certificate (not secure) | False |
+    | Use system proxy settings | False |
+
+4. Click **Test** to validate the URLs, token, and connection.
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/README.md:19]: Please add the text from the description about how the authentication works to the readme before the commands section.

File: Packs/AzureStorage/Integrations/AzureStorage/README.md [hunk 13]
```diff
@@ -0,0 +1,637 @@
...
+                    "name": "Standard_GRS",
+                    "tier": "Standard"
+                },
+                "tags": {
+                    "ms-resource-usage": "azure-cloud-shell"
+                },
+                "type": "Microsoft.Storage/storageAccounts"
+            },
+            {
+                "id": "/subscriptions/subsciption_id/resourceGroups/esource_group_name/providers/Microsoft.Storage/storageAccounts/account_name",
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/README.md:221]: one output example is good enough, you can delete the rest as it makes the readme cumbersome.

File: Packs/AzureStorage/Integrations/AzureStorage/README.md (hunk 12 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/README.md:19]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/README.md (hunk 13 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/README.md:221]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/README.md (hunk 12 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/README.md:19]: I added it at the start instead the permissions

File: Packs/AzureStorage/Integrations/AzureStorage/README.md (hunk 11 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/README.md:3]: I added here the authentication, as Arseny said.

**Comment by roysagi** [comment]
@amshamah419 - could you please check the RN here, I wrote this with @yaakovpraisler so I want another opinion:
`- Added support for 202 HTTP response status code handling.`

**Comment by amshamah419** [comment]
@yaakovi @yaakovpraisler Let's use:
`- Added support to handle a 202 HTTP response status code.`

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py [hunk 14]
```diff
@@ -0,0 +1,778 @@
...
+            method='PUT',
+            url_suffix=f'/resourceGroups/{self.resource_group_name}/providers/Microsoft.Storage/storageAccounts/'
+                       f'{args["account_name"]}/blobServices/default/containers/{args["container_name"]}',
+            params={
+                'api-version': API_VERSION,
+            },
+            json_data={'properties': properties}
+        )
+
+    def storage_blob_containers_update_request(self, args):
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py:286]: we can unite this method with `storage_blob_containers_create_request` and simply pass the `method` as a n argument to the function depending from which command it came.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py [hunk 15]
```diff
@@ -0,0 +1,778 @@
...
+             'Last Modified Time'],
+        ),
+        raw_response=response
+    )
+
+
+def storage_blob_containers_delete(client, args):
+
+    client.storage_blob_containers_delete_request(args)
+    return 'The request to update the managed cluster was sent successfully.'
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py:697]: where did you get this line?

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 16]
```diff
@@ -0,0 +1,997 @@
...
+      type: Number
+    - contextPath: AzureStorage.BlobServiceProperties.sku.name
+      description: The SKU name.
+      type: String
+    - contextPath: AzureStorage.BlobServiceProperties.sku.tier
+      description: The SKU tier.
+      type: String
+  - arguments:
+    - default: false
+      description: The name of the storage account, required.
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:842]:
```suggestion
      description: The name of the storage account.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 17]
```diff
@@ -0,0 +1,997 @@
...
+      isArray: false
+      name: account_name
+      required: true
+      secret: false
+    - default: false
+      description: The name of the container, required.
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:848]:
```suggestion
      description: The name of the container.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 18]
```diff
@@ -0,0 +1,997 @@
...
+      predefined:
+        - Blob
+        - Container
+        - None
+      required: false
+      secret: false
+    deprecated: false
+    description: |-
+      Run this command to create or update an spesific
+      blob container
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:931]:
```suggestion
      blob container.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 19]
```diff
@@ -0,0 +1,997 @@
...
+    execution: false
+    name: azure-storage-blob-containers-update
+  - arguments:
+    - default: false
+      description: The name of the storage account, required.
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:936]:
```suggestion
      description: The name of the storage account.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 20]
```diff
@@ -0,0 +1,997 @@
...
+      secret: false
+    - default: false
+      description: Specified maximum number of containers that can be included in
+        the list.
+      isArray: false
+      name: maxpagesize
+      required: false
+      secret: false
+    deprecated: false
+    description: Run this command to get the all or spesific blob container details
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:966]:
```suggestion
    description: Run this command to get all or specific blob container details.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 21]
```diff
@@ -0,0 +1,997 @@
...
+    execution: false
+    name: azure-storage-blob-containers-list
+  - arguments:
+    - default: false
+      description: The name of the storage account, required.
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:971]:
```suggestion
      description: The name of the storage account.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 22]
```diff
@@ -0,0 +1,997 @@
...
+      isArray: false
+      name: account_name
+      required: true
+      secret: false
+    - default: false
+      description: The name of the container, required.
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:977]:
```suggestion
      description: The name of the container.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 23]
```diff
@@ -0,0 +1,997 @@
...
+      isArray: false
+      name: container_name
+      required: true
+      secret: false
+    deprecated: false
+    description: Run this command to delete a spesific blob container.
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:983]:
```suggestion
    description: Run this command to delete a specific blob container.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 24]
```diff
@@ -0,0 +1,997 @@
...
+    name: azure-storage-blob-containers-create
+  - arguments:
+    - default: false
+      description: The name of the storage account, required.
+      isArray: false
+      name: account_name
+      required: true
+      secret: false
+    - default: false
+      description: The name of the container, required.
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:895]:
```suggestion
      description: The name of the container.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 25]
```diff
@@ -0,0 +1,997 @@
...
+      secret: false
+    deprecated: false
+    description: |-
+      Run this command to create or update an spesific
+      blob container
+    execution: false
+    name: azure-storage-blob-containers-create
+  - arguments:
+    - default: false
+      description: The name of the storage account, required.
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:889]:
```suggestion
      description: The name of the storage account.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 26]
```diff
@@ -0,0 +1,997 @@
...
+    - default: false
+      description: how long this blob can be restored.
+      isArray: false
+      name: restore_policy_days
+      required: false
+      secret: false
+    deprecated: false
+    description: |-
+      Run this command to set properties for
+      the blob service in a spesific account storage
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:778]:
```suggestion
      the blob service in a specific account storage.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 27]
```diff
@@ -0,0 +1,997 @@
...
+      description: The field specifies blob object tracking granularity in days, typically
+        how often the blob object should be tracked.
+      type: Number
+    - contextPath: AzureStorage.BlobServiceProperties.properties.lastAccessTimeTrackingPolicy.blobType
+      description: An array of predefined supported blob types. Only blockBlob is
+        the supported value.
+      type: String
+  - arguments:
+    - default: false
+      description: The name of the storage account, required.
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:667]:
```suggestion
      description: The name of the storage account.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 28]
```diff
@@ -0,0 +1,997 @@
...
+      isArray: false
+      name: account_name
+      required: true
+      secret: false
+    deprecated: false
+    description: Run this command to get the blob service properties of a spesific
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:637]:
```suggestion
    description: Run this command to get the blob service properties of a specific
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 29]
```diff
@@ -0,0 +1,997 @@
...
+      account storage
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:638]:
```suggestion
      account storage.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 30]
```diff
@@ -0,0 +1,997 @@
...
+      type: String
+    - contextPath: AzureStorage.StorageAccount.tags
+      description: Resource tags.
+      type: Unknown
+    - contextPath: AzureStorage.StorageAccount.type
+      description: The type of the resource.
+      type: String
+  - arguments:
+    - default: false
+      description: The name of the storage account, required.
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:631]:
```suggestion
      description: The name of the storage account
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 31]
```diff
@@ -0,0 +1,997 @@
...
+      secret: false
+    - default: false
+      description: Sets the virtual network rules
+      isArray: false
+      name: virtual_network_rules
+      required: false
+      secret: false
+    deprecated: false
+    description: |-
+      Run this command to create or update an spesific
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:463]:
```suggestion
      Run this command to create or update a specific
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 32]
```diff
@@ -0,0 +1,997 @@
...
+      account storage
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:464]:
```suggestion
      account storage.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 33]
```diff
@@ -0,0 +1,997 @@
...
+      secret: false
+    - default: false
+      description: Gets or sets the custom domain name assigned to the storage account.
+      isArray: false
+      name: custom_domain_name
+      required: false
+      secret: false
+    - auto: PREDEFINED
+      default: false
+      description: Indicates whether indirect CName validation is enabled
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:321]:
```suggestion
      description: Indicates whether indirect CName validation is enabled.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 34]
```diff
@@ -0,0 +1,997 @@
...
+      - BlobStorage
+      - FileStorage
+      - BlockBlobStorage
+      required: true
+      secret: false
+    - auto: PREDEFINED
+      default: false
+      description: Gets or sets the location of the resource. The geo region of a
+        resource cannot be changed once it is created, but if an identical geo region
+        is specified on update, the request will succeed, required.
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:268]:
```suggestion
        is specified on update, the request will succeed.
```

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml [hunk 35]
```diff
@@ -0,0 +1,997 @@
...
+      - Standard_GZRS
+      - Standard_LRS
+      - Standard_RAGRS
+      - Standard_RAGZRS
+      - Standard_ZRS
+      required: true
+      secret: false
+    - auto: PREDEFINED
+      default: false
+      description: Indicates the type of storage account, required
```
Comment (roysagi) [inline Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml:253]:
```suggestion
      description: Indicates the type of storage account.
```

Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 34 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 33 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 31 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 32 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 30 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 28 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 29 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 27 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 26 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 16 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 17 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 25 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 24 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 18 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 19 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 20 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 21 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 22 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml (hunk 23 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.yml]: done.

File: Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py (hunk 14 above)
Comment (yaakovpraisler) [reply Packs/AzureStorage/Integrations/AzureStorage/AzureStorage.py]: done.

**Comment by yaakovpraisler** [comment]
@kirbles19 please review the docs here.

**Review by roysagi — APPROVED** [review]

**Comment by ShirleyDenkberg** [comment]
@Itay4 @roysagi Submitted. In order to meet the requirements of completing all PRs within a week, there was not enough time to review the descriptions of the arguments. If required, we can revisit this at a later time.

**Pull request merged**